# **T20 World Cup 2022 Data Analysis**

In [55]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio
pio.templates.default = "plotly_white"


# Importing Dataset from Google Drive

In [56]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# Exploring Dataset

In [57]:
path = "/content/drive/MyDrive/Datasets/MensT20wcdata.csv"
df= pd.read_csv(path)
print(df.head())

   Match_id    Team_1        Team_2                                   Venue  \
0         1   Namibia     Sri Lanka  GMHBA Stadium, South Geelong, Victoria   
1         2       UAE  Netherlands   GMHBA Stadium, South Geelong, Victoria   
2         3  Scotland   West Indies                  Bellerive Oval, Hobart   
3         4   Zimbave       Ireland                  Bellerive Oval, Hobart   
4         5   Namibia  Netherlands   GMHBA Stadium, South Geelong, Victoria   

         Stage  Toss_winner Toss_decision Score_of_first_innings  \
0  First Round    Sri Lanka         Field                    163   
1  First Round          UAE           Bat                    111   
2  First Round  West Indies         Field                    160   
3  First Round      Ireland         Field                    174   
4  First Round      Namibia           Bat                    121   

  Wkts_in_first_innings Score_of_second_innings Wkts_in_second_innings  \
0                     7                   

In [58]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45 entries, 0 to 44
Data columns (total 14 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Match_id                 45 non-null     int64 
 1   Team_1                   45 non-null     object
 2   Team_2                   45 non-null     object
 3                Venue       45 non-null     object
 4   Stage                    45 non-null     object
 5   Toss_winner              45 non-null     object
 6   Toss_decision            45 non-null     object
 7   Score_of_first_innings   45 non-null     object
 8   Wkts_in_first_innings    45 non-null     object
 9   Score_of_second_innings  45 non-null     object
 10  Wkts_in_second_innings   45 non-null     object
 11  Winner                   45 non-null     object
 12  Won by                   45 non-null     object
 13  Player_of_the_match      45 non-null     object
dtypes: int64(1), object(13)
memory usage: 5.1+ K

In [59]:
print(df.shape)

(45, 14)


In [60]:
print(df.columns)

Index(['Match_id', 'Team_1', 'Team_2', '             Venue', 'Stage',
       'Toss_winner', 'Toss_decision', 'Score_of_first_innings',
       'Wkts_in_first_innings', 'Score_of_second_innings',
       'Wkts_in_second_innings', 'Winner', 'Won by', 'Player_of_the_match'],
      dtype='object')


In [61]:
print(df.describe(include= "all"))

         Match_id Team_1        Team_2                   Venue  Stage  \
count   45.000000     45            45                      45     45   
unique        NaN     16            15                       7      4   
top           NaN  India  Netherlands   Bellerive Oval, Hobart  Group   
freq          NaN      5             6                       9     31   
mean    23.000000    NaN           NaN                     NaN    NaN   
std     13.133926    NaN           NaN                     NaN    NaN   
min      1.000000    NaN           NaN                     NaN    NaN   
25%     12.000000    NaN           NaN                     NaN    NaN   
50%     23.000000    NaN           NaN                     NaN    NaN   
75%     34.000000    NaN           NaN                     NaN    NaN   
max     45.000000    NaN           NaN                     NaN    NaN   

       Toss_winner Toss_decision Score_of_first_innings Wkts_in_first_innings  \
count           45            45          

In [62]:
df.columns = df.columns.str.strip()
print(df.head())

   Match_id    Team_1        Team_2                                   Venue  \
0         1   Namibia     Sri Lanka  GMHBA Stadium, South Geelong, Victoria   
1         2       UAE  Netherlands   GMHBA Stadium, South Geelong, Victoria   
2         3  Scotland   West Indies                  Bellerive Oval, Hobart   
3         4   Zimbave       Ireland                  Bellerive Oval, Hobart   
4         5   Namibia  Netherlands   GMHBA Stadium, South Geelong, Victoria   

         Stage  Toss_winner Toss_decision Score_of_first_innings  \
0  First Round    Sri Lanka         Field                    163   
1  First Round          UAE           Bat                    111   
2  First Round  West Indies         Field                    160   
3  First Round      Ireland         Field                    174   
4  First Round      Namibia           Bat                    121   

  Wkts_in_first_innings Score_of_second_innings Wkts_in_second_innings  \
0                     7                   

In [63]:
print(df.isnull().sum())

Match_id                   0
Team_1                     0
Team_2                     0
Venue                      0
Stage                      0
Toss_winner                0
Toss_decision              0
Score_of_first_innings     0
Wkts_in_first_innings      0
Score_of_second_innings    0
Wkts_in_second_innings     0
Winner                     0
Won by                     0
Player_of_the_match        0
dtype: int64


# Data Cleaning

In [64]:
df['Player_of_the_match']= df['Player_of_the_match'].str.strip()

In [65]:
df['Score_of_first_innings'].unique()

array(['163', '111', '160', '174', '121', '152', '176', '153', '162',
       '148', '146', '132', '200', '112', '128', '159', '144', '79',
       '157', '_', '205', '179', '130', '167', '150', '91', '133', '117',
       '184', '185', '168', '141', '158', '127', '186', '137'],
      dtype=object)

In [66]:
df['Score_of_second_innings'].unique()

array(['108', '112', '118', '143', '122', '73', '180', '146', '141',
       '150', '133', '111', '113', '131', '160', '135', '51', '158',
       '105', '_', '101', '123', '129', '102', '147', '95', '137', '148',
       '159', '120', '145', '164', '144', '128', '115', '153', '170',
       '138'], dtype=object)

In [67]:
df['Score_of_first_innings'] = pd.to_numeric(df['Score_of_first_innings'], errors='coerce')
df['Score_of_first_innings'] = df['Score_of_first_innings'].astype('Int64')
df['Score_of_second_innings'] = pd.to_numeric(df['Score_of_second_innings'], errors='coerce')
df['Score_of_second_innings'] = df['Score_of_second_innings'].astype('Int64')

In [68]:
teams = pd.concat([df['Team_1'], df['Team_2']]).unique()
print(teams)

['Namibia' 'UAE' 'Scotland' 'Zimbave' 'Sri Lanka' 'West Indies'
 'New Zealand' 'Afghanistan' 'Ireland' 'Pakistan' 'Bangladesh'
 'South Africa' 'India' 'Australia' 'Netherlands' 'England' 'Netherlands ']


# Exploratory Data Analysis (EDA)

# Matches Played Per Team

In [69]:
team1 = df['Team_1'].value_counts()
team2 = df['Team_2'].value_counts()
team1

,count
Team_1,
India,5
Zimbave,4
New Zealand,4
Afghanistan,4
Sri Lanka,4
Scotland,3
Bangladesh,3
Australia,3
West Indies,2


In [70]:
team2

,count
Team_2,
Netherlands,6
Ireland,6
England,5
Pakistan,5
South Africa,4
Zimbave,4
Sri Lanka,4
Bangladesh,2
New Zealand,2


In [71]:

matches_played = team1.add(team2, fill_value=0).sort_values(ascending=False)
matches_played

,count
Ireland,8.0
Sri Lanka,8.0
Zimbave,8.0
Pakistan,7.0
England,7.0
Netherlands,6.0
New Zealand,6.0
India,6.0
Afghanistan,5.0
South Africa,5.0


# Number of Matches Played by each Team

In [72]:
fig = px.bar(
    x=matches_played.index,
    y=matches_played.values,
    title="Matches Played by Each Team",
    labels={'x': 'Team', 'y': 'Matches'}
)

fig.show()

# Matches Won by each Team

In [73]:
wins = df['Winner'].value_counts()
wins

,count
Winner,
England,5
Netherlands,4
India,4
Sri Lanka,4
No Result,4
Pakistan,4
Ireland,3
Zimbave,3
Australia,3


In [74]:
fig = px.bar(
    x=wins.index,
    y= wins.values,
    title= ("Matches Won by each Team "),
    labels= {"x": "Team", "y": "Wins"},
    color= wins.values
)
fig.show()

# Impact of Toss Outcome on the Match

In [75]:
toss_win_match_win = (df['Toss_winner'] == df['Winner']).sum()
total_matches = len(df)
labels= ["Won After Toss", "Lost After Toss"]
values = [toss_win_match_win, total_matches - toss_win_match_win]
fig = px.pie(
    names= labels,
    values = values,
    title = ("Impact of Toss Match Outcome")
)
fig.show()

In [76]:
figure = px.bar(df,
                x=df["Winner"],
                title="Number of Matches Won by teams in t20 World Cup 2022")
figure.show()

In [77]:
figure = px.bar(df,
                x=df["Venue"],
                title="Number of Matches Won at Differnt Stadiums")
figure.show()

In [78]:
print(df.columns)

Index(['Match_id', 'Team_1', 'Team_2', 'Venue', 'Stage', 'Toss_winner',
       'Toss_decision', 'Score_of_first_innings', 'Wkts_in_first_innings',
       'Score_of_second_innings', 'Wkts_in_second_innings', 'Winner', 'Won by',
       'Player_of_the_match'],
      dtype='object')


# Top Scorers of T20 World Cup 2022

In [79]:
figure = px.bar(df,
                x=df["Player_of_the_match"],
                y = df["Winner"],
                color = df["Winner"],
                title="Top Scorers in T20 World Cup 2022")
figure.show()

# First & Second Innings Score Comparison

In [80]:
fig = px.scatter(
    df,
    x="Score_of_first_innings",
    y="Score_of_second_innings",
    color="Winner",
    title="Score Comparison"
)

fig.show()

# Batting first vs Chasing

In [81]:
df['First_Innings_Win'] = df['Score_of_first_innings'] > df['Score_of_second_innings']

bat_first_wins = df[df['First_Innings_Win']]['Winner'].count()
chasing_wins = df[~df['First_Innings_Win']]['Winner'].count()

print("Bat First Wins:", bat_first_wins)
print("Chasing Wins:", chasing_wins)

Bat First Wins: 24
Chasing Wins: 18


In [82]:
fig= px.bar(
    x= ["Bat First", "Chasing"],
    y= [bat_first_wins, chasing_wins],
    title= "Batting Fist VS Chasing Wins"

)
fig.show()

# Runs Scored in the 1st Innings

In [83]:
fig= px.histogram(
    df,
    x="Score_of_first_innings",
    nbins=15,
    title="Score Distribution of First Innings")
fig.show()

# Runs Scored in 2nd Innings

In [84]:
fig= px.histogram(
    df,
    x="Score_of_second_innings",
    nbins=15,
    title="Score Distribution of Second Innings")
fig.show()

# Group Stage Wins

In [85]:
stage_wins = df.groupby("Stage")["Winner"].value_counts().reset_index()

fig = px.bar(
    stage_wins,
    x="Stage",
    y="count",
    color="Winner",
    title="Wins by Stage"
)
fig.show()

# Score Correlation Heatmap

In [86]:
corr = df[['Score_of_first_innings', 'Score_of_second_innings']].corr()
fig = px.imshow(corr, text_auto=True, title= "Score Correlation Heatmap")
fig.show()

# Win Margin

In [87]:
df['Won by'].value_counts().head(10)


,count
Won by,
5 Wickets,7
No Result,4
6 Wickets,3
31 Runs,2
4 Wickets,2
9 Wickets,2
7 Wickets,2
42 Runs,2
5 Runs,2


In [88]:
fif= px.histogram(df, x="Won by", title="Win Margin Distribution")
fif.show()

# Highest Score in 1st Innings

In [89]:
first_score= df['Score_of_first_innings'].max()
print("Highest Score in First Innings:", first_score)

Highest Score in First Innings: 205


# Highest Score in 2nd Innings

In [90]:
first_score= df['Score_of_second_innings'].max()
print("Highest Score in Second Innings:", first_score)

Highest Score in Second Innings: 180


## T20 World Cup 2022 Data Analysis: Key Insights

After the analysis performed in this notebook, here are some key insights into the T20 World Cup 2022:

### 1. Matches Played Per Team
*   Teams like Ireland, Sri Lanka, and Zimbabwe played the most matches (8 each), indicating their participation in the initial 'First Round' stage, in addition to the 'Group' stage.
*   Major teams like India, England, and Pakistan played 6-7 matches, reaching the later stages of the tournament.

### 2. Matches Won by Each Team
*   England, the eventual champions, secured the most wins (5 matches).
*   Netherlands, India, Sri Lanka, and Pakistan also showed strong performances with 4 wins each.
*   Interestingly, 'No Result' accounted for 4 instances, suggesting rain-affected or abandoned matches during the tournament.

### 3. Impact of Toss Outcome on Match Result
*   The analysis shows that in **16 out of 45 matches**, the team that won the toss also won the match. Conversely, in 29 matches, the team that won the toss lost the match. This suggests that winning the toss does not guarantee a win, as the 'Lost After Toss' category has a higher count.

### 4. Matches Won at Different Stadiums
*   The graphs depicting wins per stadium were not detailed enough to extract specific trends, but they show the distribution of wins across various venues, with some stadiums hosting more matches and thus having more wins recorded.

### 5. Top Scorers / Player of the Match
*   The `Player_of_the_match` data shows individual performances that were crucial for their team's victories. While a specific breakdown of top scorers was not explicitly calculated as a continuous value, the plot highlights which players received the award and for which winning team.

### 6. First & Second Innings Score Comparison
*   The scatter plot between `Score_of_first_innings` and `Score_of_second_innings` with the `Winner` overlay indicates that there isn't a direct linear relationship, but it does show various score combinations leading to wins. Teams could win by scoring high in the first innings and defending, or by chasing effectively.

### 7. Batting First vs Chasing
*   **Batting first resulted in 24 wins**, while **chasing resulted in 18 wins**. This indicates a slight advantage for teams batting first in the tournament, though the difference is not overwhelmingly large, suggesting a fairly balanced competition between setting a target and chasing.

### 8. Runs Scored in the 1st and 2nd Innings
*   The histograms show the distribution of scores. The `Score_of_first_innings` histogram reveals a spread of scores, with a notable peak around the 150-180 mark. The `Score_of_second_innings` histogram similarly shows a distribution, with many successful chases falling within a similar range.

### 9. Group Stage Wins
*   The 'Wins by Stage' bar chart illustrates how teams performed in different stages (First Round, Group, Semi Final, Final). England's sole win in the 'Final' stage, and one win in the 'Semi Final' stage, along with Pakistan's, highlights their progression to the tournament's climax. The 'Group' stage had the most diverse winners.

### 10. Score Correlation Heatmap
*   There is a positive correlation of `0.336455` between `Score_of_first_innings` and `Score_of_second_innings`. This weak positive correlation suggests that while higher first innings scores might be somewhat associated with higher second innings scores (or higher targets leading to higher chase attempts), it's not a strong dependency.

### 11. Win Margin Distribution
*   The most common win margins were '5 Wickets' (7 times), followed by 'No Result' (4 times), and '6 Wickets' (3 times). This highlights that many matches were won by chasing a target, and many were closely contested.

### 12. Highest Scores
*   The **highest score in the first innings was 205**.
*   The **highest score in the second innings was 180**.